docs/best_practices.md

```diff
@@ -0,0 +1,130 @@
+content. Therefore it is important to keep these keys secured. However, as they are also
+required for adding, updating, and deleting content in your Trusted Collection, they are
+more likely to be stored on one or more of your personal computers. The Notary client requests
+a password when saving any keys to persistent storage. This password is used as one component
+in the generation of a symmetric AES 256 key that is used to encrypt your signing key in CBC
+mode. When the signing key is required, you are asked to supply the password so that the
+symmetric can can be regenerated, and the signing key decrypted.
+
+While you may be tempted to mirror SSH common practices and generate different Delegations
+keys for every computer you publish from, the provided key rotation functionalities of TUF
+make this unnecessary. However, there is no security weakness in doing so and if you use
+a large number of different computers, it may be desirable simply from a key distribution
+perspective in that you don't have to update so many systems should one key be compromised. 
+
+### Root Key
+
+Compromise of any of the other keys can be easily dealt with via a normal key rotation. The Root
+key however anchors trust and in the case of a Targets, Snapshot, or Timestamp key compromise,
+is used to perform said key rotation. While a Root key compromise can be handled via the special
+Root key rotation mechanism, the necessity of doing this should be avoided at all costs.
+Therefore, we strongly recommend that a Yubikey is used to store a copy of the Root key, and
+is itself stored in a secure location like a safe. Furthermore, a paper copy of the Root key
+should be stored in a bank vault. Note that when Notary generates a new Root key, regardless
+of whether a Yubikey is present, it will always store a password encrypted copy on your local
+filesystem specifically so that you can make these types of backups. The `notary key import`
+command will allow you to create multiple Yubikeys with the Root key loaded.
+
+Additionally, you may choose whether or not to re-use a Root key for multiple repositories.
+Notary will automatically use the first Root key it finds available, or generate a new Root
+key if none is found. By ensuring only the Root key you want to create the repository with
+is available, i.e. by attaching the appropriate Yubikey, Notary will use this key as the
+root key for a new repository.
+
+## Key Rotations
+
+If you follow the appropriate security measures, you should not need to rotate your Root
+key until the signing mechanism in use is obsoleted. We currently default to ECDSA 
+with the P256 curve and do not expect it to be obsoleted for many years. Currently though
+the Root key is published as a self signed x509 certificate with an expiry set 10 years
+in the future. Therefore, you will need to rotate your root key within this time frame.
+This time was chosen as while it shouldn't be necessary to arbitrarily rotate the Root key,
```

Comment:
this isn't perfect either, but maybe something like: "We chose a 10-year expiry as a default since the Root key shouldn't need to be rotated often, though it is still beneficial to still have this required action at some point in case you would like to update to a new key or introduce an updated cipher." ?